core/workflows/workflow_base.py

```
@@ -0,0 +1,196 @@
+
+
+@dataclass
+class WorkflowConfig:
+    """Configuration for workflows."""
+    name: str
+    batch_size: int = 32
+    num_workers: int = 4
+    use_gpu: bool = True
+    checkpoint_enabled: bool = True
+    checkpoint_interval: int = 100
+    staging_path: Path = Path("/dev/shm/workflow_staging")
```

Ruff (0.12.2)

Probable insecure usage of temporary file or directory: "/dev/shm/workflow_staging"

(S108)